### PR TITLE
Expand chain api

### PR DESCRIPTION
**Pull request by Niederb**
- In order to use `get_finalized_head()` inside of `get_finalized_block()` I had to add a trait bound for `Runtime::Header` but I think it is not too limiting.
- There is already a `genesis_hash()` in `API`. I implemented `get_genesis_block()` instead.
- Merged the `GetBlock` and `GetHeader` traits into a single `GetChainInfo` trait

### CODE REVIEW

File: examples/examples/get_blocks.rs [hunk 1]
```diff
@@ -33,16 +33,23 @@ async fn main() {
...
 
 	let header_hash = api.get_finalized_head().unwrap().unwrap();
 	println!("Latest Finalized Header Hash:\n {} \n", header_hash);
 
 	let h = api.get_header(Some(header_hash)).unwrap().unwrap();
 	println!("Finalized header:\n {:?} \n", h);
 
-	let b = api.get_signed_block(Some(header_hash)).unwrap().unwrap();
-	println!("Finalized signed block:\n {:?} \n", b);
+	let b = api.get_finalized_block().unwrap().unwrap();
```
Comment (haerdib) [inline examples/examples/get_blocks.rs:44]:
```suggestion
	let signed_block = api.get_finalized_block().unwrap().unwrap();
```

File: src/api/rpc_api/author.rs [hunk 2]
```diff
@@ -267,6 +267,7 @@ where
 	Runtime: FrameSystemConfig + GetRuntimeBlockType,
 	Runtime::RuntimeBlock: BlockTrait + DeserializeOwned,
 	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime::Header: DeserializeOwned,
```
Comment (haerdib) [inline src/api/rpc_api/author.rs:270]: Why is this necessary?

File: src/api/rpc_api/chain.rs [hunk 3]
```diff
@@ -76,6 +80,13 @@ pub trait GetBlock {
 		&self,
 		number: Option<Self::BlockNumber>,
 	) -> Result<Option<SignedBlock<Self::Block>>>;
+
+	fn get_finalized_block(&self) -> Result<Option<SignedBlock<Self::Block>>>;
```
Comment (haerdib) [inline src/api/rpc_api/chain.rs:84]: Could you add some comments? E.g. `/// Get the last finalized signed block.`

File: src/api/rpc_api/events.rs [hunk 4]
```diff
@@ -159,6 +160,7 @@ where
 	Runtime: FrameSystemConfig + GetRuntimeBlockType,
 	Runtime::RuntimeBlock: BlockTrait + DeserializeOwned,
 	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime::Header: DeserializeOwned,
```
[the remark this answers is not in the document]
Comment (Niederb) [reply src/api/rpc_api/events.rs:163]: Yes, at least at the moment because it calls `GetBlock::get_block(..)`.
I could create another `impl` with different trait bounds just for the new `get_genesis_block`, but if we merge the two traits then we will the bound on `Runtime::Header`

File: src/api/rpc_api/chain.rs [hunk 5]
```diff
@@ -116,6 +132,25 @@ where
 	) -> Result<Option<SignedBlock<Self::Block>>> {
 		self.get_block_hash(number).map(|h| self.get_signed_block(h))?
 	}
+
+	fn get_finalized_block(&self) -> Result<Option<SignedBlock<Self::Block>>> {
+		self.get_finalized_head()?
```
[the remark this answers is not in the document]
Comment (Niederb) [reply src/api/rpc_api/chain.rs:137]: Yes, I think merging the traits makes sense.

File: examples/examples/get_blocks.rs [hunk 6]
```diff
@@ -33,16 +33,23 @@ async fn main() {
...
+	println!("Finalized block:\n {:?} \n", b);
+	let last_block_number = b.block.header.number;
+	let block_numbers = std::cmp::max(0, last_block_number - 3)..last_block_number;
```
[the remark this answers is not in the document]
Comment (Niederb) [reply examples/examples/get_blocks.rs]: I made it a bit clearer and added some comments

File: src/api/rpc_api/chain.rs (hunk 3 above)
Comment (Niederb) [reply src/api/rpc_api/chain.rs:60]: Done

File: src/api/rpc_api/chain.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Niederb) [reply src/api/rpc_api/chain.rs:127]: I merged the two traits now into the new `GetChainInfo` trait

[the remark this answers is not in the document]
Comment (haerdib) [reply src/api/rpc_api/chain.rs:127]: Awesome, thanks !

**Review by haerdib — APPROVED** [review]
Thanks a lot! 

One last request: Could you change the label to E2 (breaks api) and include in the PR description that you merged the two chain traits?

**Review by echevrier — APPROVED** [review]
LGTM
